### PR TITLE
Add cross-compile instructions for Raspberry Pi

### PR DESCRIPTION
**Pull request by droark**
These are basic instructions that lead to the build process completing without error (after a known error occurs and is rectified). For now, the final results are untested.

### CODE REVIEW

**Comment by droark** [comment]
For now, this is completely untested. I do have an RPi3 but it'll be a little while before I can load this up and give it a spin. If anybody wants to test my instructions and give suggestions, I'm all ears, otherwise I'll sit on this for a bit until I have enough free time to try it out.

File: doc/INSTALL.md [hunk 1]
```diff
@@ -235,6 +236,34 @@ Build with:
       make PIE=1 DEVELOPER=0 \
       CONFIGURATOR_CC="arm-linux-androideabi-clang -static"
 
+To cross-compile for Raspberry Pi
+--------------------
+
+Obtain the [official Raspberry Pi toolchains](https://github.com/raspberrypi/tools). This document assumes compilation will occur towards the Raspberry Pi 3 (arm-linux-gnueabihf as of Mar. 2018). In addition, obtain and install cross-compiled versions of sqlite 3 and gmp.
+
```
Comment (arowser) [inline doc/INSTALL.md]: Maybe better add sqlite3 and gmp cross build guide link that you mentioned in https://github.com/ElementsProject/lightning/issues/653#issuecomment-375424387

File: doc/INSTALL.md [hunk 2]
```diff
@@ -235,6 +236,34 @@ Build with:
...
+Depending on your toolchain location and target arch, source env variables will need to be set. They can be set from the command line as such:
+
+    export PATH=$PATH:/path/to/arm-linux-gnueabihf/bin
+    # Change next line depending on specific Raspberry Pi device
+    target_host=arm-linux-gnueabihf
```
Comment (arowser) [inline doc/INSTALL.md]: if should be export target_host=arm-linux-gnueabihf?

File: doc/INSTALL.md [hunk 3]
```diff
@@ -235,6 +236,34 @@ Build with:
       make PIE=1 DEVELOPER=0 \
       CONFIGURATOR_CC="arm-linux-androideabi-clang -static"
 
+To cross-compile for Raspberry Pi
+--------------------
+
+Obtain the [official Raspberry Pi toolchains](https://github.com/raspberrypi/tools). This document assumes compilation will occur towards the Raspberry Pi 3 (arm-linux-gnueabihf as of Mar. 2018). In addition, obtain and install cross-compiled versions of sqlite 3 and gmp.
```
Comment (arowser) [inline doc/INSTALL.md]: add or change to `git clone https://github.com/raspberrypi/tools.git` for easy follow

File: doc/INSTALL.md (hunk 1 above)
Comment (droark) [reply doc/INSTALL.md]: Hi. I know this horrifically after-the-fact but I figured I should reply. :) Anyway, I think this is a good idea, although the PR has now been accepted. I'll think about whether or not I should make another one.

File: doc/INSTALL.md (hunk 2 above)
Comment (droark) [reply doc/INSTALL.md]: What I wrote works on Ubuntu 17.10. I can't comment on other OSes.

File: doc/INSTALL.md (hunk 3 above)
Comment (droark) [reply doc/INSTALL.md]: I think GH users can figure out how to clone the toolchains from the link. :) I just like it when users have access to more info than a simple Git repo address.

**Comment by Sjors** [comment]
I tried to tweak these instructions to work with Armbian (for all the other pi devices), but not much luck. Any help appreciated at #1558.

**Pull request merged**